tests/conftest.py

```diff
@@ -57,17 +57,19 @@ def mock_rootme_api_manager(mocker):
 
 
 @pytest.fixture
-def mock_dummy_db_manager(mock_rootme_api_manager):
+def mock_database_manager(mock_rootme_api_manager, monkeypatch, tmp_path):
     rootme_api_manager = mock_rootme_api_manager
+    monkeypatch.setenv("DB_FOLDER", str(tmp_path))
 
-    db = DummyDBManager(rootme_api_manager)
+    db = DatabaseManager(rootme_api_manager)
     yield db
+    db.db.close()
```

Comment:
Could you explain to me what those lines do ?
I didn't know this kind of code for tests before.

Reply:
(I wanted to select the whole function here)

Reply:
The pytest function uses `fixtures` to create temporary objects to test on before deleting them.
So, in this `fixture`, we create a temporary dummy database. Then, we yield it for the tests to be done in the `test_*` functions.
Finally, this function is called once again at the end to process the remaining lines of the function (after the `yield`) and to delete the temporary created database.